File: tests/performance/python/primes.py
#!/usr/bin/env python
import sys

def print_prime(p):
	if p % 2 == 0: 
		return

	i = 3
	while i*i < p:
		if p % i == 0:
			return
		i += 2
		
	print p

def main():
	max_prime = int(sys.argv[1])
	for i in xrange(2, max_prime):
		print_prime(i)

if __name__ == "__main__":
	main()

File: tests/performance/c/primes.c
#include <stdlib.h>
#include <stdio.h>

void print_prime(int p) {
	int i;

	if(p % 2 == 0) {
		return;
	}

	for(i = 3; i * i <= p; i = i + 2) {
		if(p % i == 0) {
			return;
		}
	}
	printf("%d\n", p);
}

int main(int argc, char const *argv[]) {
	int i;
	int max_prime = atoi(argv[1]);

	for(i = 2; i <= max_prime; i = i + 1) {
		print_prime(i);
	}

	return 0;
}